app/Providers/AppServiceProvider.php

```diff
@@ -32,5 +38,16 @@ public function register()
                 $this->app->get(Logger::class)
             );
         });
+
+        $this->app->bind(AcademicRepository::class, function (): AcademicQueryBuilderRepository {
+            return new AcademicQueryBuilderRepository(
+                $this->app->get('db'),
+                $this->app->get(CrudFacade::class),
+                new DatabaseAcademicMapper(new FromAcademicPartialMapping()),
+                $this->app->get(Logger::class)
+            );
+        });
+
```

Comment:
Use this instead:

```
$this->app->bind(AcademicRepository::class, AcademicQueryBuilderRepository::class);
$this->app->when(AcademicRepository::class)
          ->needs(AcademicMapper::class)
          ->give(function () {
              return new DatabaseAcademicMapper(new FromAcademicPartialMapping())
          });
```

Reply:
Are we using it somewhere else?
It  doesn't make sense to change it we are only using it in one please, but I'll change it to make it in order to fo faster 😄

Reply:
@cnastasi done